src/core/createEvent.js

```diff
@@ -18,43 +18,76 @@ export default () => {
   let userData;
   let expectsResponse = false;
   let documentUnloading = false;
+  let frozen = false;
+
+  const guardFrozen = func => {
```

Comment:
You are right.  This isn't necessary as we don't expose the event externally, and once the event is in the payload object, it cannot be changed.  I'll remove the freezing.  When I was working with @Aaronius to come up with a solution for this, he seemed to think it would be necessary, but I agree it is over-kill.